### PR TITLE
Refactor to make pool taint / label and context configurable

### PR DESCRIPTION
**Pull request by scrosby**
…ontext configurable

## Changes proposed in this PR

- Add configuration knobs to choose which node selector cook puts on its pods.
- Add configuration knobs to choose which taint cook uses to choose its pool name.
- Add configuration knob to select one context out of a kubeconfig file.

## Why are we making these changes?

Give us  more configurability for how we connect to k8s.

### CODE REVIEW

File: scheduler/src/cook/kubernetes/compute_cluster.clj [hunk 1]
```diff
@@ -623,6 +623,8 @@
                                   ; don't need the functionality. But we need to set the file anyways
                                   ; to avoid a NPE.
                                   (.setFile kubeconfig (File. config-file))
+                                  (when kubeconfig-context
+                                    (.setContext kubeconfig kubeconfig-context))
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/compute_cluster.clj:627]: Why is this needed?

Comment (scrosby) [reply scheduler/src/cook/kubernetes/compute_cluster.clj:627]: In cases where you have a ~/.kube/config file that has multiple contexts, it won't know which context it should use and will pick one, which might not be the one you wanted it to use. With this change, we can now explicitly select which one.

**Review by dposada — APPROVED** [review]

**Comment by dposada** [comment]
Will merge when green
